tests/conftest.py

```diff
@@ -473,6 +473,28 @@ def test_multiple_microvms(test_fc_session_root_path, context, bin_cloner_path):
         shutil.rmtree(os.path.join(test_fc_session_root_path, microvms[i].id))
 
 
+@pytest.fixture(autouse=True, scope="session")
```

Comment:
As it is this right now, will it run every time tests are ran or only when we tests under the `performance` folder?

Reply:
Tl;dr: The test is run once per each execution of pytest.

Looking at https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request it notes:

> “Autouse” fixtures are a convenient way to make all tests automatically request them

Meaning all tests become `my_test(..., test_spectre_mitigations)`.

Looking at https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session it notes:

> we can add a scope="module" parameter to the [@pytest.fixture](https://docs.pytest.org/en/6.2.x/reference.html#pytest.fixture) invocation to cause a smtp_connection fixture function, responsible to create a connection to a preexisting SMTP server, to only be invoked once per test module

and:

> Fixtures are created when first requested by a test, and are destroyed based on their scope:
> - `function`: the default scope, the fixture is destroyed at the end of the test.
> - `class`: the fixture is destroyed during teardown of the last test in the class.
> - `module`: the fixture is destroyed during teardown of the last test in the module.
> - `package`: the fixture is destroyed during teardown of the last test in the package.
> - `session`: the fixture is destroyed at the end of the test session.

Meaning this fixture (`test_spectre_mitigations`) is run once at the beginning of each test `session`, meaning each execution of pytest.

Reply:
so, if I'm developing on a machine that does not implement these mitigations and run the tests, will the tests fail, right? I thought this is something we don't want

Reply:
If we want to only run when a performance test is run, you can just move the fixture to `tests/integration_tests/performance/conftest.py`

Reply:
@bchalios 

> so, if I'm developing on a machine that does not implement these mitigations and run the tests, will the tests fail, right? I thought this is something we don't want

Yes in this instance the tests would fail. This instance is extremely rare, I do not expect it to ever occur given all modern machines implement spectrev2 mitigations.

My proffered approach here is to use the simplest solution of one `conftest.py`. In my opinion it would not be worth adding the additional complexity of another `contest.py` for this issue as I expect what you mention will never occur.

However if you have a stronger opinion here I can make this change.

Reply:
It just seems weird to me to have as a requirement to implement production-level requirements on a dev machine. The way I see it is that the test is meant to make sure that our performance tests (and baselines) are being executed consistently on a production-like machine.

Reply:
`./tools/devtool test` runs performance tests which require these mitigations to be implemented to be accurate. So in this case most of the time when running tests on a dev machine you will be running some performance tests (the tests required to merge a PR).

Notably adding a second `conftest.py` also introduces errors needing to be resolved around importing/re-exporting items, this may require some restructuring making it a non-trivial change.

Reply:
We were having discussions before to add messages for the users to let them know that performance testing is not reliable in non-supported platforms.

Anyways, we can add this now and re-visit that when/if the need arises.